stanify/builders/stan_block_builder.py

```diff
@@ -0,0 +1,673 @@
+            driving_data_ftn_set,
+            stock_initial_values: Dict[str, str]
+    ):
+        self.code = IndentedString()
+        self.code += "transformed data {\n"
+        self.code.indent_level += 1
+        variable_ast_dict: Dict[str, AbstractSyntax] = {}
+        for element in self.abstract_model.sections[0].elements:
+            stan_varname = vensim_name_to_identifier(element.name)  # if stan_varname not in driving_data_ftn_set:
+            variable_ast_dict[stan_varname] = element.components[0].ast
+
+        self.code += f"real initial_time = {self.stan_model_context.initial_time};\n"
+        self.code += "\n"
+        self.code += "// Initial ODE values\n"
+        for outcome_variable_name in integ_outcome_vector_names:
+            if outcome_variable_name in stock_initial_values:
+                continue
+            for element in self.abstract_model.sections[0].elements:
+                if vensim_name_to_identifier(element.name) == outcome_variable_name:
+                    component = element.components[0]
+                    assert isinstance(component.ast, IntegStructure), "Output variable component must be an INTEG."
+                    self.code += f"real {outcome_variable_name}__init = {InitialValueCodegenWalker(lookup_function_dict, driving_data_ftn_set, variable_ast_dict).walk(component.ast)};\n"
+                    break
+        self.code += "\n"
+
+        self.code += "// Initialize ODE stock vector\n"
+        self.code += f'vector[{len(integ_outcome_vector_names)}] initial_outcome; \n'
+        for index, name in enumerate(integ_outcome_vector_names, 1):
+            if name in stock_initial_values:
+                self.code += f"initial_outcome[{index}] = {stock_initial_values[name]};  // Defined within stan\n"
+            else:
+                self.code += f"initial_outcome[{index}] = {name}__init;\n"
+
+        self.code.indent_level -= 1
+        self.code += "}\n"
+
+        return str(self.code)
+
+
+class StanParametersBuilder:
+    def __init__(self, precision_context: "PrecisionContext", stan_model_context: "StanModelContext", vensim_model_context: "VensimModelContext"):
+        self.precision_context = precision_context
+        self.stan_model_context = stan_model_context
+        self.vensim_model_context = vensim_model_context
+
+    def build_block(self, hier_est_param_names):
+
+        data_variable_names = tuple(self.stan_model_context.stan_data.keys())
+        code = IndentedString()
+        code += "parameters{\n"
+        code.indent_level += 1  # Enter parameters block
+
+        added_parameters = set()
+
+        for statement in self.stan_model_context.sample_statements:
+            if statement.lhs_expr in data_variable_names:
+                continue
+            if statement.lhs_expr in added_parameters: #TODO @Dashadower ??
+                continue
+            if statement.distribution_type == statement.assignment_dist: #TODO @Dashadower ??
```

Comment:
> ??

Remember there are two types of statements in stan; you have a sampling statement and an assignment.
If a variables is defined using a statement, it doesn't go into the parameters block, but instead defined in transformed parameters.

This conditional is making sure that we don't define parameters if the statement type is an assignment.